mesonbuild/modules/gnome.py

```diff
@@ -971,15 +972,19 @@ def genmarshal(self, state, args, kwargs):
             raise MesonException(
                 'Sources keyword argument must be a string or array.')
 
+        new_genmarshal = meson.version_compare(self._get_native_glib_version(), '>= 2.53.3')
+
         cmd = [find_program('glib-genmarshal', output + '_genmarshal')]
         known_kwargs = ['internal', 'nostdinc', 'skip_source', 'stdinc',
-                        'valist_marshallers']
+                        'valist_marshallers', 'extra_args']
         known_custom_target_kwargs = ['build_always', 'depends',
                                       'depend_files', 'install_dir',
                                       'install_header']
         for arg, value in kwargs.items():
             if arg == 'prefix':
                 cmd += ['--prefix', value]
+            elif arg == 'extra_args' and new_genmarshal:
```

Comment:
Should probably warn if this is used on old versions rather than silently ignore?